code/go/0chain.net/smartcontract/dbs/event/user_aggregate.go

```diff
@@ -206,41 +206,77 @@ func (edb *EventDb) updateUserAggregates(e *blockEvents) error {
 		}
 	}
 
-	ids := make(map[string]interface{})
-	for i := range updatedAggrs {
-		updatedAggrs[i].Round = e.round
-		ids[updatedAggrs[i].UserID] = struct{}{}
+	idSet := make(map[string]bool)
+	for _, aggr := range updatedAggrs {
+		idSet[aggr.UserID] = true
+	}
+	uniqueIds := make([]string, 0, len(idSet))
+	for id := range idSet {
+		uniqueIds = append(uniqueIds, id)
 	}
 
-	latest, err := edb.GetLatestUserAggregates(ids)
+	// load user snapshots  
+	snaps, err := edb.GetUserSnapshotsByIds(uniqueIds)
 	if err != nil {
-		logging.Logger.Error("can't load latest aggregates", zap.Error(err))
+		logging.Logger.Error("can't load latest snapshots", zap.Error(err))
 		return err
 	}
 
+	snapsMap := make(map[string]UserSnapshot, len(updatedAggrs))
+	for _, snap := range snaps {
+		snapsMap[snap.UserID] = snap
+	}
+
 	for _, aggr := range updatedAggrs {
-		u := aggr
-		//if u.Round == e.round {
-		//	logging.Logger.Error("duplicate round", zap.Any("aggr", aggr), zap.Any("latest", latest[u.UserID]))
-		//}
-		a, ok := latest[u.UserID]
+		snap, ok := snapsMap[aggr.UserID]
 		if !ok {
-			latest[u.UserID] = &u
+			snapsMap[aggr.UserID] = UserSnapshot{
+				UserID:          aggr.UserID,
+				CollectedReward: aggr.CollectedReward,
+				PayedFees:       aggr.PayedFees,
+				TotalStake:      aggr.TotalStake,
+				ReadPoolTotal:   aggr.ReadPoolTotal,
+				WritePoolTotal:  aggr.WritePoolTotal,
+			}
 			continue
 		}
-		merge(a, &u)
+		merge(&snap, &aggr)
+		snap.UpdatedAt = time.Now()
+		snapsMap[aggr.UserID] = snap
```

Comment:
this assignment is not needed

Reply:
Needed it since `snapsMap` is a map of values not ptrs. Changed it to map of ptrs.